### ISSUE TITLE
dicm2nii user interface on MacOS

### ISSUE THREAD

**Issue by neurolabusc** [xiangruili/dicm2nii#28]
I have a problem with dicm2nii with Matlab 2019a, MacOS 10.14.6 on a 2018 quad core laptop. If I launch the dicm2nii GUI, Matlab becomes unresponsive for 25 seconds each and every time I click a checkbox, I see the busy (spinning beach ball) icon, the activity monitor shows 100% resources used. I do not see the same behavior on a similar laptop also with 2019a but with MacOS 10.15. I also do not see this for simple GUIs I create that also display a checkbox.

-Inserting Pauses [does not fix the problem](http://undocumentedmatlab.com/articles/solving-a-matlab-hang-problem)
-I am not using [Magnet, Moom, PopClip, HyperDock](https://www.mathworks.com/matlabcentral/answers/421446-r2018b-r2019a-runs-very-slow-on-macos-10-14)
- [Source control disabled, maximum Java Heap Memory](https://www.mathworks.com/matlabcentral/answers/421446-r2018b-r2019a-runs-very-slow-on-macos-10-14) does not fix the problem.

I realize this is a hard problem to diagnose, as it is specific to some computers. I also am aware that the default timeout for a D-Bus method call is 25 seconds, and this issue seems to be precisely 25 seconds long.

**Comment by xiangruili** [comment]
Thank you for the detailed debug info. If the problem is with the checkbox only, my first guess will be the `setpref` function, since that is the only function the GUI checkbox calls. To test this, could you run the following line from Matlab Command line, and see if it causes the similar delay?
````
setpref('dicm2nii_gui_para', 'lefthand', true);
````

**Comment by neurolabusc** [comment]
- Running 'setpref('dicm2nii_gui_para', 'lefthand', true);' and  'setpref('dicm2nii_gui_para', 'lefthand', false);' from the command line does not cause a delay.
 - Pressing the buttons, editing text, pull down menus do not cause a delay.
 - 'Compress' and 'SPM 3D' check boxes do not cause a delay.
 - Changing any of the checkboxes in "Preferences (also apply to command line and future sessions)" do cause a delay.
 
 Is it possible that GUI is using a different thread?

**Comment by xiangruili** [comment]
Thank you for the quick feedback. My second guess is that Matlab, together with certain OS version, may not support the way for my Callback setup. Could you try the following at Matlab Command Window?

```
% set up a simple checkbox for testing. Click it after settting up Callback
h = uicontrol(gcf, 'Style', 'checkbox', 'String', 'click me');

% set up its Callback as in dicm2nii.m. Expect to replicate the delay
h.Callback = 'fprintf(''Current value: %g\n'', get(gcbo,''Value''));';

% set up Callback by passing object. Expect to fix the problem
h.Callback = @(o,~)fprintf('Current value: %g\n', get(o,'Value'));

```

**Comment by neurolabusc** [comment]
@xiangruili none of your suggestions helped. However, I discovered that I could toggle this bug on and off by turning on and off the 'Zoom' feature in the 'Accessibility' pane of the System Preferences. Turning on the Zoom feature also disrupts the SPM12 file selection dialog. It is unfortunate, as I like the ability to dynamically zoom in and out of my screen. However, this feature is not compatible with MATLAB. Thanks for all your help and great software. I hope this discovery helps others who encounter this odd issue. I was able to elicit the bug in simpler MATLAB demos, so it is certainly not specific to dicm2nii.

For those who encounter this bug - to disable the Zoom feature make sure the mouse and keyboard Zoom options are unchecked:

![zoomer](https://user-images.githubusercontent.com/8930807/74070039-e0641480-49cd-11ea-8656-ba7eb6324534.png)
